contracts/vaults/Deposit.sol

```diff
@@ -13,14 +13,23 @@ contract Deposit is Base {
     * @param _amount Amount of pool's token to deposit. Must be at least `MINIMUM_DEPOSIT`
     **/
     function deposit(uint256 _pid, uint256 _amount) external nonReentrant {
+        if (!poolInfos[_pid].committeeCheckedIn)
+            revert CommitteeNotCheckedInYet();
         if (poolDepositPause[_pid]) revert DepositPaused();
         if (_amount < MINIMUM_DEPOSIT) revert AmountLessThanMinDeposit();
         
         //clear withdraw request
         withdrawEnableStartTime[_pid][msg.sender] = 0;
         PoolInfo storage pool = poolInfos[_pid];
         UserInfo storage user = userInfo[_pid][msg.sender];
-        _claimReward(_pid, user);
+        updatePool(_pid);
+        if (user.shares > 0) {
+            uint256 pending = user.shares * pool.rewardPerShare / 1e12 - user.rewardDebt;
+            // no need to update the rewardDebt yet because this function is nonReentrant
```

Comment:
this is not a good comment, as the reentrancy can happen by calling another fucntion (like `claimReward` which is not marked nonReentrant)

Reply:
Ok so you do want to update rewardDebt here too? Or mark claimReward nonReentrant as well?